File: solutions/python/isogram/1/isogram.py
from collections import defaultdict
def is_isogram(string: str) -> bool:
    char_count = defaultdict(int)
    for ch in string.casefold():
        if ch.isalpha():
            char_count[ch] += 1

    return not any(value > 1 for value in char_count.values())

File: solutions/cpp/isogram/1/isogram.cpp
#include "isogram.h"
#include <algorithm>
#include <cctype>
#include <map>
namespace isogram {

// TODO: add your solution here
    bool is_isogram(const std::string_view& text) {
        std::map<char, int> char_count{};
        for (const auto& ch : text) {
            if (std::isalpha(ch)) {
                ++char_count[static_cast<char>(std::tolower(ch))];
            }
        }
        
        return !std::any_of(char_count.cbegin(), char_count.cend(),
                           [](const auto& kvpair) { return kvpair.second > 1; });
    }

}  // namespace isogram
